### ISSUE TITLE
kubeletconfig error

### ISSUE THREAD

**Issue by pkhadka56** [terraform-google-modules/terraform-google-kubernetes-engine#2193]
### TL;DR

I'm using safer-cluster module, version 24.1.0. Since there was no option to deactivate the insecure readonly port, I followed GKE doc: https://cloud.google.com/kubernetes-engine/docs/how-to/disable-kubelet-readonly-port to deactivate the port at cluster level using the following command:

```
gcloud container clusters update CLUSTER_NAME \
    --location=LOCATION \
    --no-enable-insecure-kubelet-readonly-port
```

After that, I created a new node pool. Everything went well. However, on the next plan, I saw the following changes will be done

<img width="480" alt="image" src="https://github.com/user-attachments/assets/e46e2a4b-ee28-4f59-ae61-0e7574cf11b5">

Due to this, I am getting the following error.

```
Error: googleapi: Error 400: At least one of ['node_version', 'image_type', 'updated_node_pool', 'locations', 'workload_metadata_config', 'upgrade_settings', 'kubelet_config', 'linux_node_config', 'tags', 'taints', 'labels', 'node_network_config', 'gcfs_config', 'gvnic', 'confidential_nodes', 'logging_config', 'fast_socket', 'resource_labels', 'accelerators', 'windows_node_config', 'machine_type', 'disk_type', 'disk_size_gb', 'storage_pools', 'containerd_config', 'resource_manager_tags', 'performance_monitoring_unit', 'queued_provisioning', 'max_run_duration'] must be specified. Details: [ { "@type": "type.googleapis.com/google.rpc.RequestInfo", "requestId": "0x193b832d5c52beb4" } ] , badRequest
```

I'm also not able to revert the changes.

I think the `kubeletConfig` was empty before. When I ran the above gcloud command, it changed kubeletConfig as the following

```
kubeletConfig:
  insecureKubeletReadonlyPortEnabled: False
```

### Expected behavior

There should be no additional changes and the terraform plan run successfully.

### Observed behavior

Got the error as mentioned above.

### Terraform Configuration

```hcl
safer-cluster v24.1.0
terraform v1.3.8
```

### Terraform Version

```sh
terraform v1.3.8
```

### Additional information

I tried setting `insecureKubeletReadonlyPortEnabled` to True again and create a new node-pool. In the first run, there is no error but in the next run, you will get the same error. I can only set insecureKubeletReadonlyPortEnabled value to True or False and can't remove it.

**Comment by waiyanwh** [comment]
@pkhadka56 How did you solve it ? I am facing the same issue

**Comment by amozoss** [comment]
> [@pkhadka56](https://github.com/pkhadka56) How did you solve it ? I am facing the same issue

This seemed to fix it for us
```  
kubelet_config {
   cpu_manager_policy = "none" # Default value
}
```
[Source for defaults ](https://cloud.google.com/kubernetes-engine/docs/how-to/node-system-config#editing_by_updating_an_existing_node_pool)

Compared to another pool that had the exact same terraform settings and our newest one had 

```
   "kubeletConfig": {                                        
      "insecureKubeletReadonlyPortEnabled": true              
    },   
```

```
sdiff -s compute.json long-run.json  
    "maxNodeCount": 4,                                        |     "maxNodeCount": 1,
    "diskSizeGb": 200,                                        |     "diskSizeGb": 50,
    "kubeletConfig": {                                        <
      "insecureKubeletReadonlyPortEnabled": true              <
    },                                                        <
      "node_pool": "compute"                                  |       "node_pool": "long_running"
```
